### ISSUE TITLE
ngAria 1.5.7 not adding aria-required to inputs

### ISSUE THREAD

**Issue by willsteinmetz** [angular/angular.js#15252]
We are currently using ngAria at my company to get a head start on helping with accessibility. One of our designers noticed while reviewing accessibility that none of our fields marked as required were being noted as such with a screen reader. Upon inspection, we found that no input fields (regular HTML text inputs, nothing custom) were not having the `aria-required` attribute added when the `required` or `ng-required` attributes were present on the element. The problem can be seen by inspecting the text inputs in this Plunkr http://plnkr.co/edit/QXNlfaNUP5TGNB227i62 and viewing that neither has the `aria-required` attribute applied.

**Comment by Narretz** [comment]
I think this is correct because screen readers should be able to read the normal "required" attribute in this case. Aria-required should afaik only be added to custom input controls.

**Comment by willsteinmetz** [comment]
It seems to depend on the field. Sometimes it does, sometimes it doesn't. ChromeVox doesn't read it in either my company's code or the Plunkr. macOS accessibility will read the Plunkr as required sometimes but not consistently. If I manually add `aria-required` to the fields then it always does.

**Comment by Narretz** [comment]
Is there a consistent difference between ng-required and just required, or is it just random?

**Comment by willsteinmetz** [comment]
It seems random. I wish it were more consistent so that I could provide better help in diagnosing it.

**Comment by gkalpak** [comment]
This was indeed a deliberate change in v1.5.x, that shouldn't affect accessibility, since (as @Narretz mentioned) native inputs are "accessible by default". From the [Migration Guide][1]:

> Due to d06431e, the `ngAria`-enhanced directives (e.g. `ngModel`, `ngDisabled` etc) will not apply ARIA attributes to native inputs, unless necessary. Previously, ARIA attributes were always applied to native inputs, despite this being unnecessary in most cases.
> [...]
> This change will not affect the accessibility of your applications (since native inputs are accessible by default), but if you relied on ARIA attributes being present on native inputs (for whatever reason), you'll have to add and update them manually.

Also, as mentioned in the [HTML5 spec][2], it is not recommended to use `aria-*` attributes that match the _default implicit ARIA semantics_:

> In the majority of cases setting an ARIA role and/or aria-* attribute that matches the default implicit ARIA semantics is unnecessary and not recommended as these properties are already set by the browser.

--
I am going to close this since it is working as expected afaict. If you can identify a scenario where Angular does not do the right thing, I would be happy to reopen and investigate.

BTW, if you want to have the `aria-*` attributes, it shouldn't be difficult to create your own directives (even copying the 1.4.x `ngAria` ones).

[1]: https://docs.angularjs.org/guide/migration#migrating-from-1-4-to-1-5
[2]: https://www.w3.org/TR/html5/dom.html#aria-usage-note
